quiz/views.py

```diff
@@ -118,8 +118,15 @@ def _get_checked_answers(saved_answers, questions):
 
 
 def _compute_result(quiz, score):
-    results = sorted(quiz.results.all(), key=operator.attrgetter('score'))
-    for i in range(0, len(results) - 2):
-        if results[i].score < score < results[i + 1].score:
-            return results[i+1]
-    return results[len(results) - 1]
+    score_ranges = quiz.score_ranges.all()
+    if not score_ranges:
+        return None
+
+    ranges = zip(score_ranges, score_ranges[1:])
```

Comment:
That looks very nice. I like the use of zip(), but this approach still does not handle case of only 1 score range. It will always return None in the next if statement.

So I suggest you to order the queryset by score in desc order, and just iterate over it one by one while the score is smaller then the range. (This way it the score_range score would act like an lower bound of the range but might be much simpler)

Reply:
Also, at the end of the project, this function might need some tests because it has a bit of logic there.